### PR TITLE
Clarify that Slic "defers" the idle timeout by sending Ping frames from client to server.

### LINKED ISSUE #337: Connection idle timeout
On https://docs.icerpc.dev/icerpc/slic-transport/connection-idle-timeout, we document:

> It's up to the transport implementation to decide if and when Ping frames are sent.

My understanding is this sub-chapter (The Slic transport) documents both the Slic protocol and its implementation, so it's odd to read "It's up to the transport implementation".

We should document what the Slic transport actually does. There was a number of PRs in icerpc-csharp related to this topic. For example:
https://github.com/icerpc/icerpc-csharp/pull/3361

### PR DESCRIPTION
**Pull request by bernardnormier**
Fixes #337.

See also my blog post on this topic.

### CODE REVIEW

File: pages/icerpc/slic-transport/connection-idle-timeout.md [hunk 1]
```diff
@@ -5,28 +5,37 @@ description: Understand connection idle timeout.
...
+See also [InactivityTimeout].
+{% /callout %}
 
-{% callout %}
+Slic ensures a connection does not become idle by sending [Ping] frames. A Slic client sends `Ping` frame to the
+server; a Slic server never sends `Ping` frames to its client.
 
-The `Ping` frame can be used for different purposes. For example, it can be used to keep alive a connection or to
-measure the round-trip time (RTT). The opaque payload allows to identify the different `Ping` frames.
+A Slic client sends a `Ping` frame (idle timeout / 2) after connection establishment and (idle timeout / 2) after any
```
Comment (pepone) [inline pages/icerpc/slic-transport/connection-idle-timeout.md:30]: What is a Slic client? It would be clear to use Slic client connection

**Review by externl — APPROVED** [review]
Looks good. See comment.

File: pages/icerpc/slic-transport/connection-idle-timeout.md [hunk 2]
```diff
@@ -5,28 +5,37 @@ description: Understand connection idle timeout.
...
+Slic is responsible to keep a connection from becoming idle when the application doesn't write anything to this
+connection.
 
-It's up to the transport implementation to decide if and when `Ping` frames are sent.
+{% callout %}
+TODO: this behavior should be configurable and controlled by the upper protocol (e.g., `icerpc`). Currently, it isn't.
 
-A `Ping` frame carries an opaque payload. This payload is sent back with the `Pong` frame. Multiple `Ping` frames can be
-sent concurrently.
+See also [InactivityTimeout].
```
Comment (bentoi) [inline pages/icerpc/slic-transport/connection-idle-timeout.md:24]: I don't see the point of the TODO and the reference to the `icerpc` protocol inactivity timeout property.

File: pages/icerpc/slic-transport/connection-idle-timeout.md [hunk 3]
```diff
@@ -5,28 +5,37 @@ description: Understand connection idle timeout.
...
 
-To prevent the idle timeout from being triggered, the client or server can send a [Ping] frame. When the peer receives
-this `Ping` frame, it resets its idle timeout timer and send back a [Pong] frame. The reception of the `Pong` frame will
-in turn reset the idle timeout timer of the receiver.
+Slic is responsible to keep a connection from becoming idle when the application doesn't write anything to this
+connection.
 
-It's up to the transport implementation to decide if and when `Ping` frames are sent.
+{% callout %}
+TODO: this behavior should be configurable and controlled by the upper protocol (e.g., `icerpc`). Currently, it isn't.
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply pages/icerpc/slic-transport/connection-idle-timeout.md]: Removed TODO

File: pages/icerpc/slic-transport/connection-idle-timeout.md (hunk 2 above)
Comment (bernardnormier) [reply pages/icerpc/slic-transport/connection-idle-timeout.md]: Removed TODO and reference to Inactivity Timeout.

**Review by pepone — APPROVED** [review]

**Review by externl — APPROVED** [review]

**Review by bentoi — APPROVED** [review]

**Pull request merged**